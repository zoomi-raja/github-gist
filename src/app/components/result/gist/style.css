.gist {
  background-color: bisque;
  flex-basis: 30%;
  padding: 2rem;
  flex-grow: 0;
  overflow: hidden;
}
.gist__user {
  display: flex;
  align-items: center;
  margin-bottom: 1.2rem;
}
.user__name--limit {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 80%;
  overflow: hidden;
  vertical-align: middle;
}
.user__avatar {
  flex-shrink: 0;
  width: 20%;
  height: 20%;
  margin-right: 1.2rem;
}
.user__avatar img {
  border-radius: 50%;
}

.user__info {
  display: flex;
  flex-direction: column;
}
.gist__desc {
  background-color: #fff;
  overflow-x: hidden;
  overflow-y: scroll;
  text-overflow: ellipsis;
  border-radius: 3px;
  padding: 1.2rem;
  max-height: 8rem;
}
@media (max-width: 700px) {
  .gist {
    flex-basis: 100%;
  }
}
