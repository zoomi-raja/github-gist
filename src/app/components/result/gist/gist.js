import { dateToYMD } from '../../../utilities/date';
import './style.css';
export function Gist() {    
  return (
    <div className='gist'>
      <div className='gist__user'>
        <div className='user__avatar'>
          <img src='https://avatars1.githubusercontent.com/u/48676961?v=4' alt="git user"/>
        </div>
        <div className='user__info'>
          <a className='user__name' href='https://' target="_blank" rel="noopener noreferrer">
            <span>zoomi</span>
            <span>&rarr;</span>
          </a>
          <span>{dateToYMD(new Date('2020-12-29T09:56:36Z'))}</span>
        </div>
			</div>
      <div className='gist__desc'>
        <a href='https://gist.github.com/ff66629bf211eef1c59d517c9754f51b'>html_url description description</a>
      </div>
    </div>
  )
}