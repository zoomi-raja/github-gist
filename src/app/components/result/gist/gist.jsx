import { dateToYMD } from '../../../utilities/date';
import PropTypes from 'prop-types';

import './style.css';
export function Gist({data}) {    
  return (
    <div className='gist'>
      <div className='gist__user'>
        <div className='user__avatar'>
          <img src={data.owner.avatar_url} alt="git user"/>
        </div>
        <div className='user__info'>
          <a className='user__name' href={data.owner.html_url} target="_blank" rel="noopener noreferrer">
            <span className='user__name--limit' >{data.owner.login}</span>
            <span>&rarr;</span>
          </a>
          <span>{dateToYMD(new Date(data.created_at))}</span>
        </div>
			</div>
      <div className='gist__desc'>
        <a href={data.html_url} target="_blank" rel="noopener noreferrer">{data.description?data.description:'No Discription Available'}</a>
      </div>
    </div>
  )
}
Gist.propTypes = {
  data: PropTypes.object
};