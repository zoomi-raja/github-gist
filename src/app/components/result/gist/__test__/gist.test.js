import renderer from 'react-test-renderer';
import { Gist } from '../gist';
it('gist snapshot', () => {
  const data =  {
    "url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678",
    "forks_url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678/forks",
    "commits_url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678/commits",
    "id": "bf2876a3c3bd668b3596a38ef4480678",
    "node_id": "MDQ6R2lzdGJmMjg3NmEzYzNiZDY2OGIzNTk2YTM4ZWY0NDgwNjc4",
    "git_pull_url": "https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678.git",
    "git_push_url": "https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678.git",
    "html_url": "https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678",
    "files": {
      "untrusted-lvl3-solution.js": {
        "filename": "untrusted-lvl3-solution.js",
        "type": "application/javascript",
        "language": "JavaScript",
        "raw_url": "https://gist.githubusercontent.com/Untrusted-Game/bf2876a3c3bd668b3596a38ef4480678/raw/89140e2451eebe5ca5c64ab5bf8b464cf1a68867/untrusted-lvl3-solution.js",
        "size": 990
      }
    },
    "public": true,
    "created_at": "2020-12-29T12:59:32Z",
    "updated_at": "2020-12-29T12:59:32Z",
    "description": "Solution to level 3 in Untrusted: http://alex.nisnevich.com/untrusted/",
    "comments": 0,
    "user": null,
    "comments_url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678/comments",
    "owner": {
      "login": "Untrusted-Game",
      "id": 37789057,
      "node_id": "MDQ6VXNlcjM3Nzg5MDU3",
      "avatar_url": "https://avatars2.githubusercontent.com/u/37789057?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Untrusted-Game",
      "html_url": "https://github.com/Untrusted-Game",
      "followers_url": "https://api.github.com/users/Untrusted-Game/followers",
      "following_url": "https://api.github.com/users/Untrusted-Game/following{/other_user}",
      "gists_url": "https://api.github.com/users/Untrusted-Game/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Untrusted-Game/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Untrusted-Game/subscriptions",
      "organizations_url": "https://api.github.com/users/Untrusted-Game/orgs",
      "repos_url": "https://api.github.com/users/Untrusted-Game/repos",
      "events_url": "https://api.github.com/users/Untrusted-Game/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Untrusted-Game/received_events",
      "type": "User",
      "site_admin": false
    },
    "truncated": false
  }
  const tree = renderer.create(<Gist key={data.id} data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('gist empty description snapshot', () => {
  const data =  {
    "url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678",
    "forks_url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678/forks",
    "commits_url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678/commits",
    "id": "bf2876a3c3bd668b3596a38ef4480678",
    "node_id": "MDQ6R2lzdGJmMjg3NmEzYzNiZDY2OGIzNTk2YTM4ZWY0NDgwNjc4",
    "git_pull_url": "https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678.git",
    "git_push_url": "https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678.git",
    "html_url": "https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678",
    "files": {
      "untrusted-lvl3-solution.js": {
        "filename": "untrusted-lvl3-solution.js",
        "type": "application/javascript",
        "language": "JavaScript",
        "raw_url": "https://gist.githubusercontent.com/Untrusted-Game/bf2876a3c3bd668b3596a38ef4480678/raw/89140e2451eebe5ca5c64ab5bf8b464cf1a68867/untrusted-lvl3-solution.js",
        "size": 990
      }
    },
    "public": true,
    "created_at": "2020-12-29T12:59:32Z",
    "updated_at": "2020-12-29T12:59:32Z",
    "description": "",
    "comments": 0,
    "user": null,
    "comments_url": "https://api.github.com/gists/bf2876a3c3bd668b3596a38ef4480678/comments",
    "owner": {
      "login": "Untrusted-Game",
      "id": 37789057,
      "node_id": "MDQ6VXNlcjM3Nzg5MDU3",
      "avatar_url": "https://avatars2.githubusercontent.com/u/37789057?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Untrusted-Game",
      "html_url": "https://github.com/Untrusted-Game",
      "followers_url": "https://api.github.com/users/Untrusted-Game/followers",
      "following_url": "https://api.github.com/users/Untrusted-Game/following{/other_user}",
      "gists_url": "https://api.github.com/users/Untrusted-Game/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Untrusted-Game/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Untrusted-Game/subscriptions",
      "organizations_url": "https://api.github.com/users/Untrusted-Game/orgs",
      "repos_url": "https://api.github.com/users/Untrusted-Game/repos",
      "events_url": "https://api.github.com/users/Untrusted-Game/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Untrusted-Game/received_events",
      "type": "User",
      "site_admin": false
    },
    "truncated": false
  }
  const tree = renderer.create(<Gist key={data.id} data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
})