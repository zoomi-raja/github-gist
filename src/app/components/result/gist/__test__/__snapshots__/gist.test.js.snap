// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gist empty description snapshot 1`] = `
<div
  className="gist"
>
  <div
    className="gist__user"
  >
    <div
      className="user__avatar"
    >
      <img
        alt="git user"
        src="https://avatars2.githubusercontent.com/u/37789057?v=4"
      />
    </div>
    <div
      className="user__info"
    >
      <a
        className="user__name"
        href="https://github.com/Untrusted-Game"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="user__name--limit"
        >
          Untrusted-Game
        </span>
        <span>
          →
        </span>
      </a>
      <span>
        2020-12-29
      </span>
    </div>
  </div>
  <div
    className="gist__desc"
  >
    <a
      href="https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678"
      rel="noopener noreferrer"
      target="_blank"
    >
      No Discription Available
    </a>
  </div>
</div>
`;

exports[`gist snapshot 1`] = `
<div
  className="gist"
>
  <div
    className="gist__user"
  >
    <div
      className="user__avatar"
    >
      <img
        alt="git user"
        src="https://avatars2.githubusercontent.com/u/37789057?v=4"
      />
    </div>
    <div
      className="user__info"
    >
      <a
        className="user__name"
        href="https://github.com/Untrusted-Game"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="user__name--limit"
        >
          Untrusted-Game
        </span>
        <span>
          →
        </span>
      </a>
      <span>
        2020-12-29
      </span>
    </div>
  </div>
  <div
    className="gist__desc"
  >
    <a
      href="https://gist.github.com/bf2876a3c3bd668b3596a38ef4480678"
      rel="noopener noreferrer"
      target="_blank"
    >
      Solution to level 3 in Untrusted: http://alex.nisnevich.com/untrusted/
    </a>
  </div>
</div>
`;
